Tasks/Maven/maventask.ts

```diff
@@ -3,97 +3,181 @@
     var mavenPath = tl.getPathInput('mavenPath', true, true);
-    mvntool = path.join(mavenPath, 'bin/mvn');
+    mvnExec = path.join(mavenPath, 'bin/mvn');
 
+    // Set the M2_HOME variable to a custom Maven installation path?
     if (tl.getBoolInput('mavenSetM2Home')) {
-        tl.setEnvVar("M2_HOME", mavenPath);
+        tl.setEnvVar('M2_HOME', mavenPath);
         tl.debug('M2_HOME set to ' + mavenPath)
     }
-} else {
-    tl.debug('Using maven from standard system path');
-    mvntool = tl.which('mvn', true);
 }
-tl.debug('Maven binary: ' + mvntool);
-
-var mavenPOMFile = tl.getPathInput('mavenPOMFile', true, true);
-var mavenOptions = tl.getInput('options', false); // options can have spaces and quotes so we need to treat this as one string and not try to parse it
-var mavenGoals = tl.getDelimitedInput('goals', ' ', true); // This assumes that goals cannot contain spaces
-
-var mvnv = tl.createToolRunner(mvntool);
-mvnv.arg('-version');
-
-var mvnb = tl.createToolRunner(mvntool);
-mvnb.arg('-f');
-mvnb.pathArg(mavenPOMFile);
-mvnb.argString(mavenOptions);
-mvnb.arg(mavenGoals);
-
-// update JAVA_HOME if user selected specific JDK version or set path manually
-var javaHomeSelection = tl.getInput('javaHomeSelection', true);
-var specifiedJavaHome = null;
+else {
+    // mavenVersionSelection is set to 'Default'
+    
+    // First, look for Maven in the M2_HOME variable
+    var m2HomeEnvVar: string = null;
+    m2HomeEnvVar = tl.getVariable('M2_HOME');
+    if (m2HomeEnvVar) {
+        tl.debug('Using M2_HOME environment variable value for Maven path: ' + m2HomeEnvVar);
+        mvnExec = path.join(m2HomeEnvVar, 'bin/mvn');
+    }
+    // Second, look for Maven in the system path
+    else {
+        tl.debug('M2_HOME environment variable is not set, so Maven will be sought in the system path');
+        mvnExec = tl.which('mvn', true);
+    }
+}
+tl.debug('Maven executable: ' + mvnExec);
 
+// Set JAVA_HOME to the JDK version (default, 1.7, 1.8, etc.) or the path specified by the user
+var specifiedJavaHome: string = null;
 if (javaHomeSelection == 'JDKVersion') {
-    tl.debug('Using JDK version to find and set JAVA_HOME');
-    var jdkVersion = tl.getInput('jdkVersion');
-    var jdkArchitecture = tl.getInput('jdkArchitecture');
+    // Set JAVA_HOME to the specified JDK version (default, 1.7, 1.8, etc.)
+    tl.debug('Using the specified JDK version to find and set JAVA_HOME');
+    var jdkVersion: string = tl.getInput('jdkVersion');
+    var jdkArchitecture: string = tl.getInput('jdkArchitecture');
 
     if (jdkVersion != 'default') {
         // jdkVersion should be in the form of 1.7, 1.8, or 1.10
         // jdkArchitecture is either x64 or x86
         // envName for version 1.7 and x64 would be "JAVA_HOME_7_X64"
-        var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
+        var envName: string = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
```

Comment:
jdkVersion is controlled input - it's a drop down selection list, user can only pick from a few choices (right now 1.6, 1.7 and 1.8)